distribution/src/bin-filemode-755/hz-cli

```diff
@@ -46,4 +46,14 @@ $JVM_OPTIONS \
 $JAVA_OPTS \
 )
 
+# trim CLASSPATH
+CLASSPATH="${CLASSPATH##:}"
+CLASSPATH="${CLASSPATH%%:}"
+
+# Bash on Windows may produce paths such as /c/path/to/lib and Java wants c:\path\to\lib
+# and, in this case, the cygpath command *should* be available - use it to format paths
+if [ "$(command -v cygpath)" != "" ]; then
+  CLASSPATH=$(cygpath -w -p "$CLASSPATH")
+fi
+
```

Comment:
Will this work ok when CLASSPATH is defined in the environment (which gets prepended to the jars in lib)?

Reply:
This "cooks" the entire CLASSPATH as it is defined at the moment it runs. If any part of it comes from the environment, it should be cleaned too. But if any part of it is modified or appended *afterwards* then it won't be cooked. Does that answer your question?

Reply:
What if the part from the environment is already cleaned? I suppose the `CLASSPATH` env variable would come from standard windows variables, so it would already be in the correct format. We end up with a mix:

From the env:
```
CLASSPATH="c:\path\to\lib1;c:\path\to\lib2"
```

Then here
https://github.com/hazelcast/hazelcast/blob/9ecb9fd731e528a1f3c169df9e95f9eaaab2e5d2/distribution/src/bin-regular/common.sh#L66
it becomes
```
c:\path\to\lib1;c:\path\to\lib2:/c/path/to/hazelcast/lib.....
```
would that work?

Reply:
tested:
```
> CLASSPATH="c:\path\to\lib1;c:\path\to\lib2"
> CLASSPATH="$CLASSPATH:$HAZELCAST_HOME/lib:$HAZELCAST_HOME/lib/*:$HAZELCAST_HOME/bin/user-lib:$HAZELCAST_HOME/bin/user-lib/*"
> CLASSPATH=$(cygpath -w -p "$CLASSPATH")
> echo $CLASSPATH
c;C:\path\to\lib1;c;C:\path\to\lib2;C:\Program Files\Git\lib;C:\Program Files\Git\lib\*;C:\Program Files\Git\usr\bin\user-lib;C:\Program Files\Git\usr\bin\user-lib\*
```
So there *is* something wrong in the output (the `c;C:\` in the first paths) - looking into it.

Reply:
So, I have pushed a fix, which (if running on Windows) (1) ensures that CLASSPATH and CLASSPATH_DEFAULT are unix-style, trimmed and (2) combines them and (3) adds the lib/* etc and (4) converts the whole thing back to windows-style. Which means that, regardless of the format of CLASSPATH and CLASSPATH_DEFAULT, we end up with a working CLASSPATH (ie deals with the problem above).